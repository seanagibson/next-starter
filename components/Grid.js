export default ({
  children,
  gap = '10px',
  cols = '1fr',
  rows = '1fr',
  templateAreas = null,
  alignItems = 'center',
  justifyItems = 'center',
  ...props
}) => {
  const areas = templateAreas === null ? `` : `grid-template-areas: ${templateAreas}`;
  return (
    <div {...props}>
      {children}
      <style jsx>
        {`
          div {
            display: grid;
            grid-gap: ${gap};
            grid-template-columns: ${cols};
            grid-template-rows: ${rows};
            ${areas}
            align-items: ${alignItems};
            justify-items: ${justifyItems};
          }
        `}
      </style>
    </div>
  );
};
