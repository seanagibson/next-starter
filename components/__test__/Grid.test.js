import React from 'react';
import { cleanup, render } from 'react-testing-library';
import Grid from '../Grid';

afterEach(cleanup);

test('should render children', () => {
  const { getByTestId } = render(
    <Grid>
      <h1 data-testid="test-heading">Test Heading</h1>
    </Grid>
  );
  const node = getByTestId('test-heading');

  expect(node.textContent).toEqual('Test Heading');
});

test('snapshot', () => {
  const { container } = render(
    <Grid>
      <h1 data-testid="test-heading">Test Heading</h1>
    </Grid>
  );
  expect(container.firstChild).toMatchSnapshot();
});
