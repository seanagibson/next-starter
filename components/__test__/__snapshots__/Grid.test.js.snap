// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="jsx-3340093321 "
>
  <h1
    data-testid="test-heading"
  >
    Test Heading
  </h1>
</div>
`;
